examples/fabric-admin/device_manager/PairingManager.cpp

```diff
@@ -0,0 +1,147 @@
+                                                                    .SetTimeout(params->commissioningWindowTimeout)
+                                                                    .SetIteration(params->iteration)
+                                                                    .SetDiscriminator(params->discriminator)
+                                                                    .SetVerifier(params->verifier)
+                                                                    .SetSalt(params->salt)
+                                                                    .SetCallback(&self.mOnOpenCommissioningWindowVerifierCallback));
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(AppServer, "Failed to open commissioning window with verifier: %s", ErrorStr(err));
+        }
+    }
+    else
+    {
+        SetupPayload ignored;
+        CHIP_ERROR err = mWindowOpener->OpenCommissioningWindow(Controller::CommissioningWindowPasscodeParams()
+                                                                    .SetNodeId(params->nodeId)
+                                                                    .SetEndpointId(params->endpointId)
+                                                                    .SetTimeout(params->commissioningWindowTimeout)
+                                                                    .SetIteration(params->iteration)
+                                                                    .SetDiscriminator(params->discriminator)
+                                                                    .SetSetupPIN(NullOptional)
+                                                                    .SetSalt(NullOptional)
+                                                                    .SetCallback(&self.mOnOpenCommissioningWindowCallback),
+                                                                ignored);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(AppServer, "Failed to open commissioning window with passcode: %s", ErrorStr(err));
+        }
+    }
+
+    // Clean up params
+    delete params;
+}
+
+void PairingManager::OnOpenCommissioningWindowResponse(void * context, NodeId remoteId, CHIP_ERROR err, chip::SetupPayload payload)
+{
+    PairingManager * self = static_cast<PairingManager *>(context);
+    if (self->mCommissioningWindowDelegate)
+    {
+        self->mCommissioningWindowDelegate->OnCommissioningWindowOpened(remoteId, err, payload);
+        self->UnregisterOpenCommissioningWindowDelegate();
+    }
+
+    OnOpenCommissioningWindowVerifierResponse(context, remoteId, err);
+}
+
+void PairingManager::OnOpenCommissioningWindowVerifierResponse(void * context, NodeId remoteId, CHIP_ERROR err)
+{
+    LogErrorOnFailure(err);
+
+    PairingManager * self = reinterpret_cast<PairingManager *>(context);
+    VerifyOrReturn(self != nullptr, ChipLogError(NotSpecified, "OnOpenCommissioningWindowCommand: context is null"));
+}
```

Comment:
Should we be doing something here, if not can you add a comment as to why we only get if the pointer is null, Otherwise I don't understand why it is important for `PairingManager::OnOpenCommissioningWindowResponse` to call this method